From 73b2cbee59eab50882078456afe13f0be8b1bccf Mon Sep 17 00:00:00 2001
From: Chen Baozi <chenbaozi@phytium.com.cn>
Date: Fri, 14 Jul 2023 08:34:13 +0800
Subject: [PATCH 078/360] dt-bindings: Add bindings for Phytium USBHS-DRD
 controller

This patch documents the DT binding for the Phytium USBHS-DRD controller.

Signed-off-by: Wang Zhimin <wangzhimin1179@phytium.com.cn>
Signed-off-by: Chen Baozi <chenbaozi@phytium.com.cn>
Signed-off-by: Wang Yinfeng <wangyinfeng@phytium.com.cn>
---
 .../devicetree/bindings/usb/phytium,usb2.yaml | 41 +++++++++++++++++++
 1 file changed, 41 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/usb/phytium,usb2.yaml

diff --git a/Documentation/devicetree/bindings/usb/phytium,usb2.yaml b/Documentation/devicetree/bindings/usb/phytium,usb2.yaml
new file mode 100644
index 000000000000..caf762039f04
--- /dev/null
+++ b/Documentation/devicetree/bindings/usb/phytium,usb2.yaml
@@ -0,0 +1,41 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/usb/phytium,usb2.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Phytium USBHS-DRD controller bindings
+
+maintainers:
+  - Chen Baozi <chenbaozi@phytium.com>
+
+properties:
+  compatible:
+    const: phytium,usb2
+
+  reg:
+    items:
+      - description: USB controller registers
+      - description: PHY registers
+
+  interrupts:
+    maxItems: 1
+
+  dr_mode:
+    enum: [host, otg, peripheral]
+
+required:
+  - compatible
+  - reg
+  - interrupts
+
+additionalProperties: false
+
+examples:
+  - |
+    usb2_0: usb2@31800000 {
+      compatible = "phytium,usb2";
+      reg = <0x0 0x31800000 0x0 0x80000>,
+            <0x0 0x31990000 0x0 0x10000>;
+      interrupts = <GIC_SPI 32 IRQ_TYPE_LEVEL_HIGH>;
+    };
-- 
2.20.1

